File: src/main/webapp/views/anouncement/js/anouncementService.js
angular.module("AnouncementServiceModule",[])
	.factory("AnouncementService",["$resource","$http",function($resource,$http)
	{
		var anouncementService = $resource("../AnouncementController/:method", {});
		anouncementService.saveAnouncement=function(newAnouncement,successcb,errorcb)//传入的两个方法
		{
			 $http({
                 method: "POST",
                 url: "../anouncementController/saveAnouncement",
                 data: newAnouncement
             }).success(successcb).error(errorcb);
		};
		
		anouncementService.findAnouncementById=function(id,successcb,errorcb)//传入的两个方法
		{
			 $http({
				 method: "POST",
                 headers: {'Content-type': 'application/x-www-form-urlencoded'},
                 url: "../anouncementController/findAnouncementById",
                 data: {'id':id},
                 transformRequest:function (data) {return $.param(data);}
             }).success(successcb).error(errorcb);
		};
		
		anouncementService.deleteAnouncement=function(id,successcb,errorcb)//传入的两个方法
		{
			 $http({
				 method: "POST",
                 headers: {'Content-type': 'application/x-www-form-urlencoded'},
                 url: "../anouncementController/deleteAnouncement",
                 data: {'id':id},
                 transformRequest:function (data) {return $.param(data);}
             }).success(successcb).error(errorcb);
		};
		
		anouncementService.updateAnouncement=function(anouncement,successcb,errorcb)//传入的两个方法
		{
			 $http({
                 method: "POST",
                 url: "../anouncementController/updateAnouncement",
                 data: anouncement
             }).success(successcb).error(errorcb);
		};
		
		anouncementService.findAllAnouncements=function(successcb,errorcb)//传入的两个方法
		{
			$http({
				method: "POST",
				headers: {'Content-type': 'application/x-www-form-urlencoded'},
				url: "../anouncementController/findAllAnouncements",
			}).success(successcb).error(errorcb);
		};
		
		anouncementService.publish=function(anouncement,successcb,errorcb)//传入的两个方法
		{
			$http({
				method: "POST",
	            url: "../anouncementController/publish",
	            data: anouncement
			}).success(successcb).error(errorcb);
		};
		
		anouncementService.getSearchPageList=function(pageIndex,pageSize,searchCondition,successcb,errorcb)//传入的两个方法
		{
			$http({
				method: "POST",
				headers: {'Content-type': 'application/x-www-form-urlencoded'},
				url: "../anouncementController/getSearchPageList",
				data: {'pageIndex':pageIndex,'pageSize':pageSize,"searchCondition":searchCondition},
				transformRequest:function (data) {return $.param(data);}
			}).success(successcb).error(errorcb);
		};
		
		return anouncementService;
    
}]);

File: src/main/java/com/frame/service/anouncementService/AnouncementService.java
package com.frame.service.anouncementService;

import java.util.List;

import javax.servlet.http.HttpServletRequest;

import com.frame.entity.anouncement.Anouncement;

public interface AnouncementService {
	public boolean  saveAnouncement(Anouncement anouncement, HttpServletRequest request);
	
	public boolean deleteAnouncement(Integer id, HttpServletRequest request);
	
	public Anouncement findAnouncementById(Integer id, HttpServletRequest request);
	
	public boolean updateAnouncement(Anouncement anouncement, HttpServletRequest request);
	
	public List<Anouncement> findAllAnouncements(HttpServletRequest request);

	public boolean updateAnouncementState(String state, HttpServletRequest request);

	public List<Anouncement> findAnouncementByState(String state);

	public boolean publish(Anouncement anouncement, HttpServletRequest request);

	public boolean updateAnouncementStateById(Integer id, String state);

	public List<Anouncement> getSearchPageList(Integer pageIndex, Integer pageSize,
			String searchCondition);

	public int getSearchTotalItems(String searchCondition);
	
}
